ioos_qc/qartod.py

```diff
@@ -636,33 +648,37 @@ def attenuated_signal_test(inp : Sequence[N],
         input data is flagged together.
     """
 
-    assert isfixedlength(threshold, 2)
-    threshold = span(*reversed(sorted(threshold)))
+    if check_type == 'std':
+        check_func = np.std
+    elif check_type == 'range':
+        check_func = np.ptp
+    else:
+        raise ValueError('Check type "{}" is not one of ["std", "range"]'.format(check_type))
 
+    tinp = mapdates(tinp)
     with warnings.catch_warnings():
         warnings.simplefilter("ignore")
         inp = np.ma.masked_invalid(np.array(inp).astype(np.floating))
 
     # Save original shape
     original_shape = inp.shape
-    inp = inp.flatten()
 
-    if check_type == 'std':
-        check_val = np.std(inp)
-    elif check_type == 'range':
-        check_val = np.ptp(inp)
-    else:
-        raise ValueError('Check type "{}" is not defined'.format(check_type))
+    # Start with everything as not tested (0)
+    flag_arr = np.full((inp.size,), QartodFlags.UNKNOWN)
 
-    # Start with everything as passing (1)
-    flag_arr = np.ma.ones(inp.size, dtype='uint8')
-
-    if check_val < threshold.maxv:
-        flag_arr.fill(QartodFlags.FAIL)
-    elif check_val < threshold.minv:
-        flag_arr.fill(QartodFlags.SUSPECT)
-
-    # If the value is masked set the flag to MISSING
+    if test_period:
+        series = pd.Series(inp.flatten(), index=tinp.flatten())
+        windows = series.rolling(f'{test_period}s', min_periods=min_obs)
+        check_val = windows.apply(check_func, raw=True)
+    else:
+        # applying np.ptp to Series causes warnings, this is a workaround
+        series = inp.flatten()
+        check_val = np.ones_like(flag_arr) * check_func(series)
+
+    flag_arr[check_val >= suspect_threshold] = QartodFlags.GOOD
+    flag_arr[check_val < suspect_threshold] = QartodFlags.SUSPECT
+    flag_arr[np.isnan(check_val)] = QartodFlags.SUSPECT
```

Comment:
@jessicaaustin @yosoyjay Jessica is saying this should be set to UNKNOWN, I think I agree.